src/main/java/iterator/RichIterator.java

```diff
@@ -66,15 +98,22 @@ default Optional<A> nextOptional() {
      * @param f the Consumer
      */
     default void foreach(Consumer<? super A> f) {
-        throw new NotImplementedException();
+        while (this.hasNext()){
+            f.accept(this.next());
+        }
     }
 
     /**
      * @param elem checks if the iterator has the element
      * @return true if exists, false otherwise
      */
     default boolean contains(A elem) {
-        throw new NotImplementedException();
+        while (this.hasNext()) {
+            if (elem.equals(this.next())) {
+                return true;
+            }
+        }
+        return false;
```

Comment:
This is a code duplication, can you use a different function already implemented for this operation?